src/Sets/Star.jl

```diff
@@ -0,0 +1,211 @@
+export Star,
+       center,
+       basis,
+       predicate
+
+"""
+    Star{N, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}, PT<:AbstractPolyhedron{N}} <: AbstractStar{N}
+
+Type that represents a generalized star set where the predicate is polyhedral, i.e.
+
+```math
+X = \\{x ∈ \\mathbb{R}^n : x = x₀ + \\sum_{i=1}^m α_i v_i,~~\\textrm{s.t. } P(α) = ⊤ \\},
+```
+where ``x₀ ∈ \\mathbb{R}^n`` is the center, the ``m`` vectors ``v₁, …, vₘ`` form
+the basis of the star set, and ``P : α ∈ \\mathbb{R}^m → \\{⊤, ⊥\\}`` where
+``P(α) = ⊤`` if and only if ``Aα ≤ b`` is the polyhedral predicate.
+
+### Fields
+
+- `c` -- vector that represents the center
+- `V` -- matrix where each column corresponds to a basis vector
+- `P` -- polyhedral set that represents the predicate
+
+### Notes
+
+The predicate function is implemented as a conjunction of linear constraints, i.e.
+a subtype of `AbstractPolyhedron`. By a slight abuse of notation, the *predicate*
+is also used to denote the subset of ``\\mathbb{R}^n`` such that ``P(α) = ⊤`` holds.
+
+The ``m`` basis vectors (each one ``n``-dimensional) are stored as the columns
+of an ``n × m`` matrix.
+
+### Examples
+
+This example is drawn from Example 1 in [2]. Consider the two-dimensional plane
+``\\mathbb{R}^2``. Let
+
+```jldoctest star_constructor
+julia> V = [[1.0, 0.0], [0.0, 1.0]]
+2-element Array{Array{Float64,1},1}:
+ [1.0, 0.0]
+ [0.0, 1.0]
```

Comment:
I would suggest to not print such trivial intermediate results. They are easy to read from the statements and just clutter the text.